conda_build/deprecations.py

```diff
@@ -236,8 +281,8 @@ def _get_module(self, stack: int) -> tuple[ModuleType, str]:
         import inspect  # expensive
 
         try:
-            frame = inspect.stack()[2 + stack]
-            module = inspect.getmodule(frame[0])
+            frame = sys._getframe(2 + stack)
```

Comment:
Why this change? Is `sys._getframe` guaranteed to be stable? According to [the docs](https://docs.python.org/3/library/sys.html#sys._getframe), it's a CPython implementation detail (e.g. it might be absent in PyPy?). So if we are doing this for performance, we should fall back to `inspect.stack` if unavailable.

```
getframe = getattr(sys, "_getframe", None)
frame = frame(2+stack) if getframe else inspect.stack()[2+stack]
```

Reply:
Ditto, also see https://github.com/conda/conda/pull/12919

Reply:
Ok, since we are catching AttributeError it's ok I guess.